pkg/inventory/inventorycm.go

```diff
@@ -115,10 +119,31 @@ func (icm *ConfigMap) GetObject() (*unstructured.Unstructured, error) {
 	return invCopy, nil
 }
 
-func buildObjMap(objMetas object.ObjMetadataSet) map[string]string {
+func buildObjMap(objMetas object.ObjMetadataSet, objStatus []actuation.ObjectStatus) map[string]string {
 	objMap := map[string]string{}
+	objStatusMap := map[object.ObjMetadata]actuation.ObjectStatus{}
+	for _, status := range objStatus {
+		objStatusMap[ObjMetadataFromObjectReference(status.ObjectReference)] = status
+	}
 	for _, objMetadata := range objMetas {
-		objMap[objMetadata.String()] = ""
+		if status, found := objStatusMap[objMetadata]; found {
+			objMap[objMetadata.String()] = stringFrom(status)
+		} else {
+			objMap[objMetadata.String()] = ""
```

Comment:
I think we might want to return an error here if there are objects in the spec that aren't in the status.

Reply:
Done